src/asm/section.c

```diff
@@ -489,15 +504,11 @@ void sect_AlignPC(uint8_t alignment, uint16_t offset)
 		if ((sym_GetPCValue() - offset) % alignSize)
 			error("Section's fixed address fails required alignment (PC = $%04" PRIx32
 			      ")\n", sym_GetPCValue());
-	} else if (sect->align != 0) {
-		if ((((sect->alignOfs + curOffset) % (1 << sect->align)) - offset) % alignSize) {
-			error("Section's alignment fails required alignment (offset from section start = $%04"
-				PRIx32 ")\n", curOffset);
-		} else if (alignment > sect->align) {
-			sect->align = alignment;
-			sect->alignOfs = (offset - curOffset) % alignSize;
-		}
-	} else {
+	} else if (sect->align != 0 && (((sect->alignOfs + curOffset) % (1 << sect->align))
+					- offset) % alignSize) {
+		error("Section's alignment fails required alignment (offset from section start = $%04"
+			PRIx32 ")\n", curOffset);
+	} else if (alignment > sect->align) {
```

Comment:
This is just a refactoring to avoid saying the same case twice, not changing any behavior.